src/main/scala/de/upb/cs/swt/delphi/instanceregistry/daos/DatabaseAuthDAO.scala

```diff
@@ -57,28 +59,63 @@ class DatabaseAuthDAO (configuration : Configuration) extends AuthDAO with AppLo
     }
   }
 
-  override def addUser(delphiUser : DelphiUser) : Try[Long] = {
+  override def addUser(delphiUser : DelphiUser) : Try[String] = {
     if(hasUserWithUsername(delphiUser.userName)){
       Failure(new RuntimeException(s"username ${delphiUser.userName} is already exist."))
     } else {
       val id = 0L //Will be set by DB
       val userName = delphiUser.userName
       val secret = delphiUser.secret
-      val userType = delphiUser.userType
+      val userType = delphiUser.userType.toString
 
       val addFuture: Future[Long] = dbAuth.run((users returning users.map(_.id)) += (id, userName, hashString(secret), userType))
       val userId = Await.result(addFuture, Duration.Inf)
 
       log.info(s"Added user ${delphiUser.userName} with id $userId to database.")
-      Success(userId)
+      Success(userName)
+    }
+
+  }
+
+  override def removeUser(id: Long) : Try[Unit] = {
+    if(hasUserWithId(id)) {
+      removeUserWithId(id)
+      Success(log.info(s"Successfully removed user with id $id."))
+    }else{
+      val msg = s"Cannot remove user with id $id, that id is not present."
+      log.warning(msg)
+      Failure(new RuntimeException(msg))
+    }
+  }
+
+  override def getUserWithId(id: Long): Option[DelphiUser] = {
+    if(hasUserWithId(id)) {
+      val result = Await.result(dbAuth.run(users.filter(_.id === id).result.headOption), Duration.Inf)
+      Some(dataToObjectUser(result))
+    } else {
+      None
     }
+  }
 
+  override def getAlllUser(): List[DelphiUser] = {
```

Comment:
Spelling error, there's on extra *"l"* :smile:  Please fix that and the usages of course.